src/primitives/transaction.cpp

```diff
@@ -67,18 +67,18 @@ uint256 CTransaction::ComputeHash() const
     return SerializeHash(*this, SER_GETHASH, SERIALIZE_TRANSACTION_NO_WITNESS);
 }
 
-uint256 CTransaction::GetWitnessHash() const
+std::unique_ptr<const uint256> CTransaction::ComputeWitnessHash() const
 {
     if (!HasWitness()) {
-        return GetHash();
+        return nullptr;
     }
-    return SerializeHash(*this, SER_GETHASH, 0);
+    return std::unique_ptr<const uint256>{new uint256{SerializeHash(*this, SER_GETHASH, 0)}};
```

Comment:
Could use `MakeUnique<const uint256>(SerializeHash(*this, SER_GETHASH, 0));` here